xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletResponse.java

```diff
@@ -66,9 +68,21 @@ public void sendRedirect(String redirect) throws IOException
             LOGGER.warn("Possible HTTP Response Splitting attack, attempting to redirect to [{}]", redirect);
             return;
         }
+        if (redirect.matches("[a-z0-9]+://.*")) {
+            // Full URL, check that the hostname matches
+            if (!getURLSecurityManager().isDomainTrusted(new URL(redirect))) {
```

Comment:
Do we need the "IF" above? What about wrapping in a try/catch for URLMalformedException instead?

Reply:
`sendRedirect` is very often used with relative URLs (most of the time actually) so relying on exception is not great for performances.